src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Platform/Mono/MonoDebugger.ts

```diff
@@ -0,0 +1,50 @@
+import { getAssemblyNameFromUrl } from '../DotNet';
+
+const currentBrowserIsChrome = (window as any).chrome
+  && navigator.userAgent.indexOf('Edge') < 0; // Edge pretends to be Chrome
+
+let hasReferencedPdbs = false;
+
+export function hasDebuggingEnabled() {
+  return hasReferencedPdbs && currentBrowserIsChrome;
+}
+
+export function attachDebuggerHotkey(loadAssemblyUrls: string[]) {
+  hasReferencedPdbs = loadAssemblyUrls
+    .some(url => /\.pdb$/.test(getAssemblyNameFromUrl(url, /* includeExtension */ true)));
+
+  // Use the combination shift+alt+D because it isn't used by the major browsers
+  // for anything else by default
+  const altKeyName = navigator.platform.match(/^Mac/i) ? 'Cmd' : 'Alt';
+  if (hasDebuggingEnabled()) {
+    console.info(`Debugging hotkey: Shift+${altKeyName}+D (when application has focus)`);
+  }
+
+  // Even if debugging isn't enabled, we register the hotkey so we can report why it's not enabled
+  document.addEventListener('keydown', evt => {
+    if (evt.shiftKey && (evt.metaKey || evt.altKey) && evt.code === 'KeyD') {
+      if (!hasReferencedPdbs) {
+        console.error('Cannot start debugging, because the application was not compiled with debugging enabled.');
+      } else if (!currentBrowserIsChrome) {
+        console.error('Currently, only Chrome is supported for debugging.');
+      } else {
+        launchDebugger();
+      }
+    }
+  });
+}
```

Comment:
I might have missed this when you and Rodrigo were discussing, what's the reason why we need to open a new thing?

Reply:
Do you mean why we need to open a new tab? It's so we can have a debugger UI that's connected to the ws-proxy that intercepts all the commands and injects .NET info into the results. I'm not aware of any way to make the regular dev tools (on the same tab) connect to an external V8 protocol endpoint.